packages/protocol/src/abstracts/BaseRouter.sol

```diff
@@ -255,23 +265,23 @@ abstract contract BaseRouter is SystemAccessControl, IRouter {
         if (requestor != address(this)) {
           revert BaseRouter__bundleInternal_flashloanInvalidRequestor();
         }
-        _addTokenToList(asset);
+        tokensToCheck = _addTokenToList(asset, tokensToCheck);
 
         // Call Flasher.
         flasher.initiateFlashloan(asset, flashAmount, requestor, requestorCalldata);
```

Comment:
since now the beneficiary is in memory, consider verifying requestorCalldata to check if the beneficiary passed is equal to the current beneficiary.
If not done the current fix allows attacker to have different beneficiaries between parent and child.
This was mentioned in the fix suggestion as well 
 https://www.notion.so/0xmacro/Fuji-1-Preliminary-Audit-Report-external-private-b36f908d4d3c492d8ad0a9ff9fe38b1a?pvs=4#2f4ca61ff23946b594478d72cd64144b

Reply:
I agree that once the flashloan enters again it is a clean slate _beneficiary. However, the flasher calling the router back cannot contain "malicious" logic because it was checked/verified previously. 
I will drop a brainstom to check what else could be done...
Lets consider the "value extracting" actions: which those would be, withdrawing, borrowing withdrawETH, swap. Swapping cannot be the first option, because it will revert. To withdraw or borrow, you require either a permit, which is specific for an operator-receiver. Note too, that in addition the permit also now hashes the actions with the args, so permits cannot be front-run anymore. For withdrawETH, I think the _tokenBalance change will flag stop this, because when you enter with the flashloan logic you must have pushed the funds out somehow already, but the balance has to remain the same at the end of the execution of the inner(flashloan) _bundleInternal.